packages/react/src/PageHeader/PageHeader.tsx

```diff
@@ -124,9 +143,21 @@ const Root = React.forwardRef<HTMLDivElement, React.PropsWithChildren<PageHeader
       // It is very unlikely to have a PageHeader without a TitleArea, but we still want to make sure we don't break the page if that happens.
       if (!titleArea) return
 
-      // // grab the data-size-variant attribute from the titleArea
-      const sizeVariant = titleArea.getAttribute('data-size-variant')
-      setTitleVariant(sizeVariant as string)
+      // find the title element
+      const title = Array.from(titleArea.children as HTMLCollection).find(child => {
+        return child instanceof HTMLElement && child.getAttribute('data-component') === 'PH_Title'
+      })
+
+      const styles = getComputedStyle(title as HTMLHeadingElement)
```

Comment:
I'm confused about why we need to `getComputedStyle` on the title

Reply:
I mentioned [here](https://github.com/primer/react/pull/4657/files#r1633020618) why I have this, but we should find another way so that we can remove this from the useeffect because it causes layout shift. 


@mperrotti The reason I have this solution is because to support array type overrides. 👇

```jsx
<PageHeader.Title
  sx={{
    fontSize: ['26px', '26px', 'var(--text-title-size-large, 32px)', 'var(--text-title-size-large, 32px)'],
  }}
>
  Issue Title
</PageHeader.Title>
```

When [we look for, for example, `lineHeight` in sx prop](https://github.com/primer/react/blob/main/packages/react/src/PageHeader/PageHeader.tsx#L456), we assign that value to a css variable `--custom-line-height` so that the default styles can be [overwritten](https://github.com/primer/react/blob/main/packages/react/src/PageHeader/PageHeader.tsx#L339) however when we have an array type like above, css var can't resolve that array. So we need some other way to resolve it and this is not the best way to resolve it, I know 😅 I just don't know how else we can solve this.  

I pushed another [PR](https://github.com/primer/react/pull/4658) that doesn't involve javascript but it is not great either.

I also posted a discussion to talk about overrides https://github.com/github/primer/discussions/3329 - Let me know what you think! We should maybe pair on it and chat further?

Reply:
No worries about this problem anymore, [I am closing this PR](https://github.com/primer/react/pull/4657#issuecomment-2162694294) and I'll chat you tomorrow! @mperrotti